apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +27,84 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
+      OodAppkit.dataroot.join('projects').tap do |path|
         p.mkpath unless p.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /[\w-]+\z/,
+    message: 'Name may only contain letters, digits, dashes, and underscores'
+  }
+
   attr_reader :dir
+  delegate :icon, :name, :description, to: :manifest
+
+  def initialize(attributes = {})
+    @dir            = attributes.fetch(:dir, nil).to_s
+  end
+
+  def save!
+    make_manifest
+    write_manifest
+  end
+
+  def update(attributes)
+    manifest = Manifest.load(manifest_path)
+    manifest = manifest.merge(attributes)
+    manifest.valid? ? manifest.save(manifest_path) : false
+  end
+
+  def destroy!
+    FileUtils.remove_dir(project_dataroot, force = true)
+  end
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def make_manifest
+    File.new(manifest_path, 'w+') # try this: unless Dir.pwd != Project.dataroot
+  end
+  
+  def manifest_path
+    File.join(configuration_directory, 'manifest.yml')
   end
 
-  def config_dir
-    Pathname.new("#{absolute_dir}/.ondemand").tap { |p| p.mkpath unless p.exist? }
+  def configuration_directory
+    Pathname.new("#{project_dataroot}/.ondemand").tap { |path| path.mkpath unless path.exist? } 
   end
 
-  def absolute_dir
+
+  def project_dataroot
     Project.dataroot.join(dir)
   end
 
-  def save!
-    true
+  def proj_name
+    proj = dir.scan(/[\w-]+\z/)
+    proj[0].titleize
   end
 
-  def destroy!
-    FileUtils.remove_dir(absolute_dir, force = true)
+  def manifest
+    @manifest ||= Manifest.load(manifest_path)
+  end
+
+  def title
+    manifest.name.empty? ? name.titleize : manifest.name
+  end
```

Comment:
I can't get the title to render in the view like the description does for some reason. The manifest updates for both, but only the description is visible when the page is reloaded after writing the manifest.